File: 0784-letter-case-permutation/0784-letter-case-permutation.py
class Solution:
    def letterCasePermutation(self, s: str) -> List[str]:
        v =[]
        for i in range(len(s)):
            if(s[i].isalpha()):
                v.append(i)
        ans = []
        n = 1<<len(v)
        for i in range(n):
            arr = [0 for x in range(len(s))]
            for j in range(len(v)):
                if(i & (1<<j)):
                    arr[v[j]]=1
            s1=""
            for x in range(len(s)):
                if(s[x].isalpha()):
                    if(arr[x]):
                        s1+= s[x].upper()
                    else:
                        s1+=s[x].lower()
                else:
                    s1+=s[x]
            ans.append(s1)
        return ans

File: 0784-letter-case-permutation/0784-letter-case-permutation.cpp
class Solution {
public:
    vector<string> letterCasePermutation(string s) {
        vector<int>v;
        for(int i=0;i<s.size();++i){
            if(isalpha(s[i])){
                v.push_back(i);
            }
        }
        vector<string> ans ;
        int n = (1<<v.size());
        for(int i =0;i<n;++i){
            string s1 = s;
            for(int j=0;j<v.size();++j){
                if(i & (1<<j)){
                    s1[v[j]]= toupper(s1[v[j]]);
                }else{
                    s1[v[j]] = tolower(s1[v[j]]);
                }
            }
            ans.push_back(s1);
        }
        return ans;
    }
};